uPortal-api/uPortal-api-platform-impl/src/main/java/org/apereo/portal/api/portlet/DelegateState.java

```diff
@@ -32,6 +33,9 @@
      *     used
      */
     public DelegateState(PortletMode portletMode, WindowState windowState) {
+        Assert.notNull(portletMode, "Argument 'portletMode' cannot be null");
```

Comment:
I have seen using Assert in org.apereo.portal.api.permissions.Activity class. Here is a link in Jasig master branch -
https://github.com/Jasig/uPortal/blob/master/uPortal-api/uPortal-api-platform-impl/src/main/java/org/apereo/portal/api/permissions/ActivityImpl.java

Thats the reason I added it here. If you pass the null, assertion fails and it throws IllegalArgumentException. I personally think we can throw IllegalArgumentException instead of putting Assert statement in the main java classes. I would be happy to make that change.